code-challenges.test.js

```diff
@@ -14,16 +14,42 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
+
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+ describe ("aboutThem" = () => {
+   const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
-
+   it ("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", ()=>{
+    expect(aboutThem(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]);
+   })
+ })
 
 // b) Create the function that makes the test pass.
 
+// this got me dizzy..
+// iterate over the array of objects
+// access the values of name and occupation from the objects
+// TO CAPITALIZE THE NAME: since there might be 2 sets of strings, i can split the name and access the first index and capitalize it and join the rest of the words with slice... and eventually join the words again
+// make a newArray to push the complete sentence in
+
+const aboutThem = (arr) => {
+  let newArr = [];
+  for (var i = 0; i < arr.length; i++) {
+    let occupation = arr[i].occupation;
+    let name = arr[i].name.split(" ").map((value) => {
+      return value.charAt(0).toUpperCase() + value.slice(1);
+    });
+    newArr[i] = name.join(" ") + " is " + occupation;
```

Comment:
I think the issue with it only working on the first object is here `newArr[i] =`. You might try `newArr.push(...)